content/en/docs/concepts/storage/volumes.md

```diff
@@ -462,92 +462,155 @@ and then removed entirely in the v1.26 release.
 
 | Value | Behavior |
 |:------|:---------|
-| | Empty string (default) is for backward compatibility, which means that no checks will be performed before mounting the hostPath volume. |
+| `‌` | Empty string (default) is for backward compatibility, which means that no checks will be performed before mounting the `hostPath` volume. |
 | `DirectoryOrCreate` | If nothing exists at the given path, an empty directory will be created there as needed with permission set to 0755, having the same group and ownership with Kubelet. |
 | `Directory` | A directory must exist at the given path |
 | `FileOrCreate` | If nothing exists at the given path, an empty file will be created there as needed with permission set to 0644, having the same group and ownership with Kubelet. |
 | `File` | A file must exist at the given path |
 | `Socket` | A UNIX socket must exist at the given path |
-| `CharDevice` | A character device must exist at the given path |
-| `BlockDevice` | A block device must exist at the given path |
+| `CharDevice` | _(Linux nodes only)_ A character device must exist at the given path |
+| `BlockDevice` | _(Linux nodes only)_ A block device must exist at the given path |
 
-Watch out when using this type of volume, because:
+{{< caution >}}
+The `FileOrCreate` mode does **not** create the parent directory of the file. If the parent directory
+of the mounted file does not exist, the pod fails to start. To ensure that this mode works,
+you can try to mount directories and files separately, as shown in the
+[`FileOrCreate` example](#hostpath-fileorcreate-example) for `hostPath`.
+{{< /caution >}}
 
-* HostPaths can expose privileged system credentials (such as for the Kubelet) or privileged APIs
-  (such as container runtime socket), which can be used for container escape or to attack other
-  parts of the cluster.
-* Pods with identical configuration (such as created from a PodTemplate) may
-  behave differently on different nodes due to different files on the nodes
-* The files or directories created on the underlying hosts are only writable by root. You
-  either need to run your process as root in a
-  [privileged Container](/docs/tasks/configure-pod-container/security-context/) or modify the file
-  permissions on the host to be able to write to a `hostPath` volume
+Some files or directories created on the underlying hosts might only be
+accessible by root. You then either need to run your process as root in a
+[privileged container](/docs/tasks/configure-pod-container/security-context/)
+or modify the file permissions on the host to be able to read from
+(or write to) a `hostPath` volume.
 
 #### hostPath configuration example
 
-```yaml
+{{< tabs name="hostpath_examples" >}}
+{{< tab name="Linux node" codelang="yaml" >}}
+---
+# This manifest mounts /data/foo on the host as /foo inside the
+# single container that runs within the hostpath-example-linux Pod.
+#
+# The mount into the container is read-only.
```

Comment:
````suggestion

The manifest mounts `/data/foo` on the host as `/foo` inside the
single container that runs within the hostpath-example-linux Pod.
The mount into the container is read-only.

```yaml
---
````

Reply:
Why's this better? I can't guess.

Reply:
It is "better" because it would help the downstream localization. Lemme explain it in details.

Suppose we have a code snippet (in this case, the whole tab is a snippet) like this:

````
```yaml
# some comments
the:
  real:
    stuff
```
````

What we the localization team is expected to translate the comment line. We do translations by commenting out the original English text (using HTML comment syntax that is acceptable to Markdown as well) and then adding translated texts. Comments in code snippets are difficult to handle, since they are supposed to be rendered in a verbatim way. For example we cannot do this:

````
```yaml
<!-- some comments -->
# 一些注釋
the:
  real:
    stuff
```
````

We had to comment out the whole snippet out and add a new one, even if there is only one line of English comment there. That is:

````
<!--
```yaml
# some comments
the:
  real:
    stuff
```
-->
```yaml
# 一些注釋
the:
  real:
    stuff
```
````

This is kind of annoying. If those inline comments are extracted into standalone paragraphs, the localization becomes very easy. We simply don't need to touch the code snippet. We can focus on the texts instead. I mean the following case:

````
General introductions to the following YAML goes here, then the example:
```yaml
the:
  yaml:
    or: whatever code
```
````

can be handled cleanly as:

````
<!--
General introductions to the following YAML goes here, then the example:
-->
這裡是對下面的 YAML 的一般性介紹，不要寫成行內注釋：

```yaml
the:
  yaml:
    or: whatever code
```
````

Hopefully this explains why I am kinda picky on inline comments unless necessary.

Reply:
I think the snippet works better how I've proposed it.

If the overall localizations subproject recommends a particular approach then fine, let's do it.
If just the Chinese localization team prefer it - I think that isn't strong enough grounds to revise this PR. I'm wary of informally adopting style guide changes that suit single localizations. Formal adoption with discussion _is_ fine.

Reply:
@sftim Are you willing to revise the structure as suggested?
The suggested layout doesn't hurt readability, but it at least makes the downstream localization easier.

Reply:
OK, I'll have a go.

Reply:
How's the revised PR?